mtp_api/apps/transaction/views.py

```diff
@@ -1,37 +1,39 @@
-from django.forms import Form
+from django.db import transaction
+from django.http import HttpResponseRedirect
 
-from rest_framework import mixins
-from rest_framework import viewsets
-from rest_framework import filters
+import django_filters
+from rest_framework import mixins, viewsets, filters, status, exceptions
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
 
-from django_filters import FilterSet
-
-from .models import Transaction
 from mtp_auth.models import PrisonUserMapping
 from prison.models import Prison
-from .serializers import TransactionSerializer
+from .models import Transaction
+from .serializers import TransactionSerializer, \
+    CreditedOnlyTransactionSerializer
+from .constants import TRANSACTION_STATUS, TAKE_LIMIT, \
+    DEFAULT_SLICE_SIZE
+from .permissions import IsOwner, IsOwnPrison
 
 
-class TransactionFilterForm(Form):
+class StatusChoiceFilter(django_filters.ChoiceFilter):
 
-    def clean_upload_counter(self):
-        upload_counter = self.cleaned_data.get('upload_counter')
-        if upload_counter == None:
-            # get the latest if 'upload_counter' param not specified
-            try:
-                transaction = Transaction.objects.latest('upload_counter')
-                upload_counter = transaction.upload_counter
-            except Transaction.DoesNotExist:
-                pass
-        return upload_counter
+    def filter(self, qs, value):
+
+        if value in ([], (), {}, None, ''):
```

Comment:
that breaks on 0

On Fri, Jul 3, 2015 at 3:43 PM, Steve Marshall <notifications@github.com>
wrote:

> In mtp_api/apps/transaction/views.py
> <https://github.com/ministryofjustice/money-to-prisoners-api/pull/9#discussion_r33868345>
> :
>
> >
> > -    def clean_upload_counter(self):
> > -        upload_counter = self.cleaned_data.get('upload_counter')
> > -        if upload_counter == None:
> > -            # get the latest if 'upload_counter' param not specified
> > -            try:
> > -                transaction = Transaction.objects.latest('upload_counter')
> > -                upload_counter = transaction.upload_counter
> > -            except Transaction.DoesNotExist:
> > -                pass
> > -        return upload_counter
> > +    def filter(self, qs, value):
> > +
> > +        if value in ([], (), {}, None, ''):
>
> Any reason this isn't if not value?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ministryofjustice/money-to-prisoners-api/pull/9/files#r33868345>
> .
>